composeApp/src/commonMain/kotlin/com/github/familyvault/ui/screens/main/addFamilyMember/AddMemberToFamilyGroupQrCodeScanScreen.kt

```diff
@@ -23,15 +30,33 @@ class AddMemberToFamilyGroupQrCodeScanScreen : Screen {
         val navigator = LocalNavigator.currentOrThrow
         val qrCodeScanner = koinInject<IQRCodeService>()
         val coroutineScope = rememberCoroutineScope()
+        var invalidQrCode by remember { mutableStateOf(false) }
+        var scanning by remember { mutableStateOf(true) }
 
-        LaunchedEffect(Unit) {
-            coroutineScope.launch {
-                val payload = qrCodeScanner.scanPayload()
-                navigator.replaceAll(AddMemberToFamilyGroupBackendOperationsScreen(payload))
+        LaunchedEffect(scanning) {
+            if (scanning) {
+                coroutineScope.launch {
+                    try {
+                        val payload = qrCodeScanner.scanPayload()
+                        navigator.replaceAll(AddMemberToFamilyGroupBackendOperationsScreen(payload))
+                    } catch (e: CancellationException) {
+                        navigator.pop()
+                    } catch (e: Exception) {
```

Comment:
I jak będziemy mieli te wyjątki stricte dla kodów QR, to unikniemy takiego problemu, że błąd niezwiązany ze skanowaniem kodów QR zostanie zinterpretowany jako zły kod QR